promql/parse.go

```diff
@@ -104,7 +132,16 @@ func ParseMetric(input string) (m labels.Labels, err error) {
 	defer parserPool.Put(p)
 	defer p.recover(&err)
 
-	return p.parseGenerated(START_METRIC).(labels.Labels), nil
+	parseResult := p.parseGenerated(START_METRIC)
+	if parseResult != nil {
+		m = parseResult.(labels.Labels)
+	}
+
+	if len(p.parseErrors) != 0 {
+		err = p.parseErrors
```

Comment:
You should return explicitly rather than using named returns where possible, it's easier to understand.

Reply:
There is still the `defer recover()` mechanism in place, which catches parser (now unwanted) panics. As far as I know, that does only work with named returns.

Reply:
At least be consistent with the above function, and mention the values on line 144.